test/hotspot/jtreg/serviceability/jvmti/vthread/MethodExitTest/libMethodExitTest.cpp

```diff
@@ -197,12 +197,8 @@ breakpoint_hit2(jvmtiEnv *jvmti, JNIEnv* jni,
                 jboolean is_virtual, char* mname) {
   jvmtiError err;
 
-  // Verify that we did not get a METHOD_EXIT events when enabled on the cthread.
-  if (received_method_exit_event) {
-    passed = JNI_FALSE;
-    received_method_exit_event = JNI_FALSE;
-    LOG("FAILED: got METHOD_EXIT event on the cthread: %p\n", cthread);
-  }
+  // need to reset this value after the breakpoint_hit1
+  received_method_exit_event = JNI_FALSE;
```

Comment:
There was a loom-dev email thread regarding this last year. Seems related. I had concluded that the way the test was written that no MethodExit event should have been received. I'm not sure if I missed something in my analysis or if this failure is a result of your changes:

https://mail.openjdk.org/pipermail/loom-dev/2023-August/006059.html
https://mail.openjdk.org/pipermail/loom-dev/2023-September/006170.html

Reply:
Thank you for the comment and links to the discussion. In fact, I've observed the MethodExit events really posted between the breakpoint hits: `hit1` and `hit2`. The first one is at the return from the `unmount()` method. I was not able to prove why they should not be expected.

Reply:
I'm not sure I follow the test logic. Its summary says "Verifies that MethodExit events are delivered on both carrier and virtual threads", but now it just ignores MethodExit requested for carrier thread in breakpoint_hit1.
Then there is no sense to request the event on carrier thread.
Per the test summary I'd expect the test should test MethodExit for carrier thread, but then java part needs to force unmount